tests/jballerina-unit-test/src/test/resources/test-src/types/future/future_negative.bal

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+
+public function testWaitTwiceOnSingleFuture() {
+    future<int|error> f4 = start calculate(9,8);
+    int|error r = wait f4 ;
+    int|error r1 = wait f4 ;
+    if(r1 is error) {
+        panic r1;
+    }
+}
+
+public function testMultipleWaitTwiceOnTwoFutures() {
+    future<int|error> f4 = start calculate(9, 8);
+    future<int|error> f5 = start calculate(5, 4);
+    record { int|error r1; int|error r2; } result = wait {r1: f4, r2: f5};
+    record { int|error r1; int|error r2; } result1 = wait {r1: f4, r2: f5};
+    int|error r1 = result1.r1;
+    if(r1 is error) {
+        panic r1;
+    }
+}
+
+public function testMultipleWaitTwiceOnDifferentFutures() {
+    future<int> f1 = start calculate(3, 4);
+    future<int|error> f4 = start calculate(9, 8);
+    future<int|error> f5 = start calculate(5, 4);
+    record { int|error r1; int|error r2; } result = wait {r1: f4, r2: f5};
+    record { int|error r1; int|error r2; } result1 = wait {r1: f4, r2: f1};
```

Comment:
Shall we check the  r2 value, which should not be an error?

Reply:
I have re written the tests to check this as well